### DISCUSSION TITLE
How to "or queries"? Desperate

### DISCUSSION THREAD

**Discussion by hulkthedev** [directus/directus#14365] [Archive]
Super simple thing, maximally complicated documented.

I have a datetime column, not necessarily filled. How do I correctly concatenate my OR query? Sorry but the documentation here isn't any help for me after only GraphQL examples or examples with real values and no "null"

`filter={"brand":{"_eq":"foo"},"language_code":{"_eq":"en"},"from_date":{"_lte":"2022-07-11 12:00:00"},"to_date":{"_gte":"2023-12-12 23:59:59"},"status":{"_eq":"published"}}`

"to_date" is not always filled, where and how do I put the "or" query in?

**Comment by benhaynes** [comment]
CC @erondpowell

**Comment by hulkthedev** [comment]
Okay, with `"to_date":{"_gte":"%now%"}` I get 2 from 4 records, they have a datetime value in that column.

With "`to_date":{"_null":"true"}` I get the other 2 datasets.
How do I link this in my filter statement, please?

**Reply by hulkthedev** [reply]
https://stackoverflow.com/questions/57867151/how-to-do-conditionally-filtering-with-directus-javascript-sdk
`"to_date":{"_gte":"now"},"to_date":{"logical":"or","_null":"true"}`

Yeah, dont work.

**Reply by benhaynes** [reply]
See Rijk's answer below. You are not using `now` correctly based on his link to Dyanmic Variables.

**Reply by hulkthedev** [reply]
Is this supposed to be "help"?
For real?

"See @docu"
"You are not using now correctly based on his link to Dynamic Variables."

This is NO help guys.

Help would be to supplement my filter, which I have posted here umpteen times, with the correct filter and thus help me instead of referring to the documentation, which I have been sitting on alone for 5 hours today. Why do you think I am writing here? If the documentation helped me, I wouldn't open a thread here.

**Reply by benhaynes** [reply]
This is not a free support forum, and you should certainly not expect responses after only hours of posting your question. We linked to the docs that provide the answer, but despite you saying you've read it "100x times", you are not following them. Your response is dissonant with our community guidelines, so I will indeed lock this post.

For anyone else who comes here looking for the "answer", be sure to use `$NOW` ... not `now`.

**Comment by rijkvanzanten** [comment]
See https://docs.directus.io/reference/filter-rules/#filter-rules 🙂
